### ISSUE TITLE
running out of drive space

### ISSUE THREAD

**Issue by robtfm** [Swatinem/rust-cache#158]
i find my workflows on github sometimes run out of drive space when using the cache action. 

i don't think the cached target folder should be larger than building directly, so i wonder if maybe the cache files are not deleted after downloading?

**Comment by max-sixty** [comment]
Not a maintainer, but do you have a reproducible example?

**Comment by robtfm** [comment]
Here’s a run which failed repeatedly, then the next commit succeeded after removing the cache action: https://github.com/decentraland/bevy-explorer/actions/runs/5798430222

**Comment by max-sixty** [comment]
Yeah, we have had a disk space issue with windows before on GHA, though not related to cache.

Something curious about the run is that a) the cache size is huge — 2.5GB, b) on the run where it [succeeded](https://github.com/decentraland/bevy-explorer/actions/runs/5798774772/job/15717279544), it only takes 6.5 minutes to compile.

Might be worth exploring why the cache is so large...

**Comment by NobodyXu** [comment]
I have an idea based on https://github.com/Swatinem/rust-cache/issues/126#issuecomment-1707841117 : 

rust-cache could compress and store the cache as a squashfs.

Then on restore, simply mount the squashfs image and use overlayfs to pass the writes to a tmpfs, and the Linux kernel would decompress each file on demand and then simply dropped the decompressed file on out-of-memory situation.
